src/core/utils/BrowserFileSystem.ts

```
@@ -0,0 +1,221 @@
+  }
+
+  static dirname(path: string): string {
+    const segments = path.split("/").filter(Boolean);
+    segments.pop();
+    return "/" + segments.join("/");
+  }
+
+  static basename(path: string): string {
+    return path.split("/").filter(Boolean).pop() || "";
+  }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)